File: uploadservice/src/main/java/net/gotev/uploadservice/http/impl/HurlBodyWriter.java
package net.gotev.uploadservice.http.impl;

import net.gotev.uploadservice.http.BodyWriter;

import java.io.IOException;
import java.io.OutputStream;

/**
 * @author Aleksandar Gotev
 */

public class HurlBodyWriter extends BodyWriter {

    private OutputStream mOutputStream;

    public HurlBodyWriter(OutputStream outputStream) {
        mOutputStream = outputStream;
    }

    @Override
    public void write(byte[] bytes) throws IOException {
        mOutputStream.write(bytes);
    }

    @Override
    public void write(byte[] bytes, int lengthToWriteFromStart) throws IOException {
        mOutputStream.write(bytes, 0, lengthToWriteFromStart);
    }

    @Override
    public void flush() throws IOException {
        mOutputStream.flush();
    }
}

File: uploadservice/src/main/java/net/gotev/uploadservice/http/hurl/HurlBodyWriter.kt
package net.gotev.uploadservice.http.hurl

import net.gotev.uploadservice.http.BodyWriter

import java.io.IOException
import java.io.OutputStream

/**
 * @author Aleksandar Gotev
 */

class HurlBodyWriter(private val stream: OutputStream) : BodyWriter() {
    @Throws(IOException::class)
    override fun write(bytes: ByteArray) {
        stream.write(bytes)
    }

    @Throws(IOException::class)
    override fun write(bytes: ByteArray, lengthToWriteFromStart: Int) {
        stream.write(bytes, 0, lengthToWriteFromStart)
    }

    @Throws(IOException::class)
    override fun flush() {
        stream.flush()
    }

    @Throws(IOException::class)
    override fun close() {
        stream.close()
    }
}
